### DISCUSSION TITLE
Delete or "hide" a (public) package

### DISCUSSION THREAD

**Discussion by hb0** [community/community#24788] [Apps, API and Webhooks]
<p>I’m testing the “new” Github Packages feature, currently with Gradle.</p>
<p>I succeeded to publish a package but want to delete this test version again.</p>
<p>The documentation says deletion is not possible but I delete all linked versions.</p>
<p>I did not find an option in the web UI of Github so I tried the API:</p>
<pre><code>curl -X POST \
-H "Accept: application/vnd.github.package-deletes-preview+json" \
-H "Authorization: bearer ACCESS_TOKEN" \
-d '{"query":"mutation { deletePackageVersion(input:{packageVersionId:\"PACKAGE_ID==\"}) { success }}"}' \
https://api.github.com/graphql
</code></pre>
<p>However, for this to work I need the ID of the package version.</p>
<p>I also did not find this in the web UI of Github so I tried the API again, but am failing:</p>
<pre><code>curl -X POST \
-H "Accept: application/vnd.github.package-deletes-preview+json" \
-H "Authorization: bearer ACCESS_TOKEN" \
-d "query {
  organization(login: "ORGANIZATION_ACCOUNT") {
    registryPackages {
      edges {
        node {
          name
          id
        }
      }
    }
  }
}" \
https://api.github.com/graphql
</code></pre>
<p>The API returns:</p>
<pre><code>{
  "message": "Problems parsing JSON",
  "documentation_url": "https://developer.github.com/v4"
}
</code></pre>
<p>When I try this via the GraphQL API Explorer:</p>
<pre><code>query {
    organization(login:"ORGANIZATION_ACCOUNT") {
    registryPackages(first: 100) {
      edges {
        node {
          name
          id
        }
      }
    }
  }
}
</code></pre>
<p>I receive:</p>
<pre><code>{
  "errors": [
    {
      "type": "INSUFFICIENT_SCOPES",
      "locations": [
        {
          "line": 6,
          "column": 11
        }
      ],
      "message": "Your token has not been granted the required scopes to execute this query. The 'name' field requires one of the following scopes: ['read:packages'], but your token has only been granted the: ['read:gpg_key', 'read:org', 'read:public_key', 'read:repo_hook', 'repo', 'user'] scopes. Please modify your token's scopes at: https://github.com/settings/tokens."
    },
    {
      "type": "INSUFFICIENT_SCOPES",
      "locations": [
        {
          "line": 7,
          "column": 11
        }
      ],
      "message": "Your token has not been granted the required scopes to execute this query. The 'id' field requires one of the following scopes: ['read:packages'], but your token has only been granted the: ['read:gpg_key', 'read:org', 'read:public_key', 'read:repo_hook', 'repo', 'user'] scopes. Please modify your token's scopes at: https://github.com/settings/tokens."
    }
  ]
}
</code></pre>
<p>If this is really the “easiest way” to accomplish my task - how can I provide a token to the GraphQL Explorer to be able to fix my query format error which the API reports unspecificly?</p>

**Comment by leomp12** [comment]
<p>Hey,</p>
<p>You can check package versions with:</p>
<pre><code>$ curl -sL -X POST https://api.github.com/graphql \
-H "Authorization: bearer &lt;github_token&gt;" \
-d '{"query":"query{repository(owner:\"&lt;repo_owner&gt;\",name:\"&lt;repo_name&gt;\"){registryPackages(first:10){nodes{packageType,registryPackageType,name,nameWithOwner,id,versions(first:10){nodes{id,version,readme}}}}}}"}' | jq .
</code></pre>
<p>But <strong>“removing a package from a public registry is not allowed”</strong> anyway (I’m having the same problem :/)</p>
<p>Ref.: <a href="https://stackoverflow.com/questions/59103177/how-to-delete-remove-unlink-unversion-a-package-from-the-github-package-registry" rel="nofollow noopener">https://stackoverflow.com/questions/59103177/how-to-delete-remove-unlink-unversion-a-package-from-the-github-package-registry</a></p>

**Comment by hb0** [comment, marked as answer]
<p>See me answer there for a “workaround” - if this is possible in your case as it comes with some potential problems. In my case the project was relativily fresh created so it was ok.</p>
<p><a href="https://stackoverflow.com/a/59105581/5815054" rel="nofollow noopener">https://stackoverflow.com/a/59105581/5815054</a></p>

**Comment by leomp12** [comment]
<p>Wow <a class="mention" href="/hb0">@hb0</a> , it works! haha</p>
<p>You’re my hero!</p>

**Comment by roel1975** [comment]
<p>Please don’t delete anything, because everything can be evidence</p>
